### PR TITLE
Rename `std::fs::try_exists` to  `std::fs::exists` and stabilize fs_try_exists

### LINKED ISSUE #83186: Tracking Issue for fs_try_exists
Feature gate: `#![feature(path_try_exists)]`

This is a tracking issue for `try_exists()` method on `std::path::Path`.

This method is similar to `exists()` method except it does **not** silently ignore errors that made it impossible to find out if the path actually exists (e.g. lack of permission on parent dir). Thus it naturally returns `io::Result<bool>` instead of just `bool`.

### Public API

```rust
mod fs {
    pub fn try_exists<P: AsRef<Path>>(path: P) -> Result<bool>;
}

impl Path {
    #[stable(since = "1.63.0")]
    pub fn try_exists(&self) -> io::Result<bool>;
}
```

### Steps / History

- https://github.com/rust-lang/rust/pull/81822
- https://github.com/rust-lang/rust/pull/85060
- https://github.com/rust-lang/rust/pull/97912

### Unresolved Questions

- None yet.

### PR DESCRIPTION
**Pull request by eduardosm**
FCP completed in tracking issue.

Tracking issue: https://github.com/rust-lang/rust/issues/83186

Closes https://github.com/rust-lang/rust/issues/83186

Stabilized API:

```rust
mod fs {
    pub fn exists<P: AsRef<Path>>(path: P) -> io::Result<bool>;
}
```

### CODE REVIEW

**Comment by simulified** [comment]
sup
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
```

File: library/std/src/path.rs [hunk 1]
```diff
@@ -2888,6 +2888,8 @@ impl Path {
     /// prevent time-of-check to time-of-use (TOCTOU) bugs. You should only use it in scenarios
     /// where those bugs are not an issue.
     ///
+    /// This is an alias to [`fs::exists`].
```
Comment (Amanieu) [inline library/std/src/path.rs:2891]:
```suggestion
    /// This is an alias for [`std::fs::exists`].
```

Comment (eduardosm) [reply library/std/src/path.rs]: Done

**Comment by fmease** [comment]
https://github.com/rust-lang/rust/pull/126252#issuecomment-2159941744
@bors r-

**Pull request merged**